### PR TITLE
FS-101 Add report director

### PR DESCRIPTION
**Pull request by mic-smith**
## Description

Create Report Director to handle file uploads

## Changelog
- Move file_upload_service to file_utils
- Add report director which uses fle_utils and returns hardcode report string

### CODE REVIEW

File: backend/src/report_director.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    filename: str | None
+    report: str | None
+
+async def report_on_file_upload(upload:UploadFile) -> FileUploadReport:
+
+    file = handle_file_upload(upload)
+
+    update_scratchpad(result=file["content"])
+
+    report = "#Report on upload as markdown" # await report_agent.invoke(file["content"])
```
Comment (stevenhillcox-sl) [inline backend/src/report_director.py:19]: I am wondering if the director should be responsible for storing the report in the scratchpad, or if the agent should be doing it.

Comment (mic-smith) [reply backend/src/report_director.py:19]: Probably the director, though I don't think we have any reason to be using the scratch pad, so it's tempting to just remove completely for now.

File: backend/src/report_director.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
[the remark this answers is not in the document]
Comment (IMladjenovic) [reply backend/src/report_director.py:1]: and if we rename the director, could be worth renaming the current supervisor to `chat_supervisor.py`

[the remark this answers is not in the document]
Comment (mic-smith) [reply backend/src/report_director.py]: Done, and renamed prompt file too to match. (`director` -> `chat_director`)

**Review by IMladjenovic — APPROVED** [review]

File: backend/src/directors/report_director.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    filename: str | None
+    report: str | None
+
+async def report_on_file_upload(upload:UploadFile) -> FileUploadReport:
+
+    file = handle_file_upload(upload)
+
+    update_scratchpad(result=file["content"])
+
+    report = "#Report on upload as markdown" # await report_agent.invoke(file["content"])
```
Comment (evpearce) [inline backend/src/directors/report_director.py:19]: are we supposed to have this bit of commented out code?
and i this report just a placeholder or something? what is it used for

Comment (stevenhillcox-sl) [reply backend/src/directors/report_director.py:19]: I believe this is the part where we will plug in the report agent once it's complete. For now, it just returns some placeholder markdown

Comment (mic-smith) [reply backend/src/directors/report_director.py:19]: Yes, this is just a sample response and to show where the report agent will be integrated. The placeholder report isn't used by anything yet.

**Pull request merged**